sunpy/net/dataretriever/sources/goes.py

```diff
@@ -52,33 +92,75 @@ def _get_url_for_timerange(self, timerange, **kwargs):
         ----------
         timerange: sunpy.time.TimeRange
             time range for which data is to be downloaded.
-        satellite_number : int
-            GOES satellite number (default = 15)
-        data_type : string
-            Data type to return for the particular GOES satellite. Supported
-            types depend on the satellite number specified. (default = xrs_2s)
+        Physobs : string
+            'INTENSITY', 'PARTICLE_FLUX' and 'IRRADIANCE', Case sensitive
+        Instrument: string
+            Fixed argument, 'goes' Case insensitive.
         """
-        # find out which satellite and datatype to query from the query times
-        sat_num = self._get_goes_sat_num(timerange.start, timerange.end)
-        base_url = 'http://umbra.nascom.nasa.gov/goes/fits/'
-        total_days = (timerange.end - timerange.start).days + 1 
-        all_dates = timerange.split(total_days)
-        result = list()
-#       Earlier we assumed that an entire range (consequently all dates within the range) were assigned the
-#       same GOES sat number, which meant we supplied timerange.start and timerange.end to
-#       _get_goes_sat_num. But now, we iterate over all the dates individually and hence
-#       _get_goes_sat_num gets each single date as an argument (the timerange.start and timerange.end
-#       for a single day is the day itself (trivial) ). That is why the same date. Its iterating over all the dates
-#       indivually.
-        for day in all_dates:
-            regex = "{date:%Y}/go{sat:02d}"
-            if (day.end < parse_time('1999/01/15')):
-                regex += "{date:%y%m%d}.fits"
-            else:
-                regex += "{date:%Y%m%d}.fits"
-            url = base_url + regex.format(
-                    date=day.end, sat=self._get_goes_sat_num(day.end, day.end)[0])
-            result.append(url)
```

Comment:
I'm confused. This code has not been removed, if you look [here in particle flux code](https://github.com/sudk1896/sunpy/blob/3c9ccf2a5dc46c98b7d625791210d054234f4b37/sunpy/net/dataretriever/sources/goes.py#L112), it is checking if it is more than three days old and then downloading this old data. You told me it would come under X-ray flux, I also checked inside the fits file [here](http://umbra.nascom.nasa.gov/goes/fits/2014/go1520140101.fits), If you look carefully you would find TTYPE2 = 'FLUX' somewhere, I didn't find irradiance anywhere hence I put this in Particle flux downloader.